src/main/java/Duke.java

```diff
@@ -1,10 +1,158 @@
+            switch (command) {
+            case COMMAND_TODO_WORD:
+                addNewTodo(commandArgs);
+                break;
+            case COMMAND_DEADLINE_WORD:
+                addNewDeadline(commandArgs);
+                break;
+            case COMMAND_EVENT_WORD:
+                addNewEvent(commandArgs);
+                break;
+            case COMMAND_LIST_WORD:
+                listItems();
+                break;
+            case COMMAND_DONE_WORD:
+                markTaskAsDone(commandArgs);
+                break;
+            default:
+                printErrorMessage();
+                printGuideMessage();
+                break;
+            }
+        } catch (Exception e) {
+                printErrorMessage();
+        }
+    }
+
+    public static String[] splitCommandWordAndArgs(String rawUserInput) {
+        final String[] split = rawUserInput.trim().split(" ", 2);
+        return split.length == 2 ? split : new String[] { split[0] , "" }; // else case: no parameters
+    }
+
+    public static void addNewListItem(Task item) {
+        list[itemsInList] = item;
+        itemsInList++;
+    }
+
+    public static void addNewTodo(String args) {
+        Task t = new ToDo(args);
+        addNewListItem(t);
+        echoNewlyAddedItem(t);
+    }
+
+    public static void addNewDeadline(String args) {
+        String description = args.substring(0, args.indexOf("\\by"));
+        String by = args.substring(args.indexOf("\\by")+3);
+        Task t = new Deadline(description, by);
+        addNewListItem(t);
+        echoNewlyAddedItem(t);
+    }
+
+    public static void addNewEvent(String args) {
+        String description = args.substring(0, args.indexOf("\\at"));
+        String at = args.substring(args.indexOf("\\at")+3);
+        Task t = new Event(description, at);
+        addNewListItem(t);
+        echoNewlyAddedItem(t);
+    }
+
+    public static void listItems() {
+        if (itemsInList > 0) {
+            System.out.println("\t Here are the tasks in your list:");
+            for (int i=0;i<itemsInList;i++) {
+                System.out.printf("\t %d. %s\n", i+1, list[i].toString());
+            }
+        } else {
+            System.out.println("\t Your list is still empty! Start adding tasks now :)");
+        }
+        printLineBreak();
+    }
+
+    public static void echoNewlyAddedItem(Task item) {
+        System.out.println("\t Got it! I've added this task:");
+        System.out.println("\t   " + item.toString());
+        System.out.printf("\t Now you have " + itemsInList + " task%s in the list.\n", (itemsInList>1 ? "s":""));
+        printLineBreak();
+    }
+
+    public static void markTaskAsDone(String listNumber) {
+        int taskID = Integer.parseInt(listNumber) - 1;
+        if (0 <= taskID && taskID < itemsInList) {
+            list[taskID].markAsDone();
+        } else {
+            printErrorMessage();
+        }
+    }
+
     public static void main(String[] args) {
```

Comment:
You can consider adding spaces in this code block similar to the rest of the code